File: src/java/me/indexyz/strap/define/AbstractContext.java
package me.indexyz.strap.define;

import me.indexyz.strap.object.Update;
import me.indexyz.strap.utils.BotNetwork;
import me.indexyz.strap.utils.Configuration;

public class AbstractContext {
    public BotNetwork network;
    public Update update;
    public Session session;
    public Configuration configuration;

    public void copy(AbstractContext context) {
        this.network = context.network;
        this.update = context.update;
        this.session = context.session;
        this.configuration = context.configuration;
    }
}

File: src/kotlin/me/indexyz/strap/define/context/AbstractContext.kt
package me.indexyz.strap.define.context

import me.indexyz.strap.`object`.Update
import me.indexyz.strap.utils.Configuration
import me.indexyz.strap.utils.Network

open class AbstractContext(
        var network: Network,
        var update: Update,
        var configuration: Configuration
)